core_backend/app/llm_call/parse_input.py

```diff
@@ -128,14 +170,20 @@ async def _identify_language(
             IdentifiedLanguage.get_prompt(),
             litellm_model=LITELLM_MODEL_LANGUAGE_DETECT,
         )
-        if identified_lang in IdentifiedLanguage.get_supported_languages():
+
+        try:
             question.original_language = getattr(IdentifiedLanguage, identified_lang)
-        else:
-            question.original_language = IdentifiedLanguage.UNKNOWN
+        except AttributeError:
```

Comment:
catches cases where llm outputs an unsupported language label e.g. KOREAN
then we catch those here, say it's `UNSUPPORTED`

Reply:
Nice.